File: piles/simulating_recursion1.cc
#include <iostream>
using namespace std;

void work(int n) {
    if (n > 0) {
        cout << ' ' << n;
        work(n - 1);
        work(n - 1);
    }
}

int main() {
    int n;
    while (cin >> n) {
        work(n);
        cout << endl;
    }
    return 0;
}

File: piles/simulating_recursion1.py
from yogi import tokens

def work(n: int) -> None:
    stack = [n]
    while stack:
        n = stack.pop()
        if n > 0:
            print('', n, end='')
            stack.append(n - 1)
            stack.append(n - 1)

def main() -> None:
    for n in tokens(int):
        work(n)
        print() 


main()